utils/clients/gemini/client_test.go

```diff
@@ -0,0 +1,136 @@
+// +build integration,vpn
+
+package gemini
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/brave-intl/bat-go/settlement"
+	"github.com/brave-intl/bat-go/utils/cryptography"
+	uuid "github.com/satori/go.uuid"
+	"github.com/shopspring/decimal"
+	"github.com/stretchr/testify/suite"
+)
+
+type GeminiTestSuite struct {
+	suite.Suite
+	secret cryptography.HMACKey
+	apikey string
+}
+
+func TestGeminiTestSuite(t *testing.T) {
+	suite.Run(t, new(GeminiTestSuite))
+}
+
+func (suite *GeminiTestSuite) SetupTest() {
+	secret := os.Getenv("GEMINI_CLIENT_SECRET")
+	apikey := os.Getenv("GEMINI_CLIENT_KEY")
+	if secret != "" {
+		suite.secret = cryptography.NewHMACHasher([]byte(secret))
+		suite.apikey = apikey
+	}
+}
+
+func (suite *GeminiTestSuite) TestBulkPay() {
+	ctx := context.Background()
+	client, err := New()
+	suite.Require().NoError(err, "Must be able to correctly initialize the client")
+	five := decimal.NewFromFloat(5)
+
+	// // uncomment if using a master key
+	// accountListRequest := suite.preparePrivateRequest(NewAccountListPayload())
```

Comment:
is there any way for us to detect whether we're using a master key?

Reply:
yes. it should be prefixed with `account-` or `master-`

Reply:
you thinking only go through this pathway if one or the other is provided?

Reply:
fixed in 9cd17a8